tests/unit/lib/Any_unittest.cc

```diff
@@ -836,3 +836,31 @@ TEST(Any, cast_into_sgobject)
 	EXPECT_EQ(any.cast<CSGObject*>(), ptr);
 	EXPECT_THROW(any.cast<int>(), std::logic_error);
 }
+
+TEST(Any, simple_visit)
+{
+	int extracted_value{};
+	struct ExtractVisitor
+	{
+	};
+	Any::register_visitor<int, ExtractVisitor>([&] (int value, ExtractVisitor* visitor) { extracted_value = value; });
```

Comment:
Just wild thining here. From api pov, I am wondering whether the template arguments could be somehow omitted and extracted from the std::function.

Reply:
Si